java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,461 @@
-            || (!prefix.equals("") && !entryName.startsWith(prefix))) {
-          File destfile = new File(entryName);
-          if (destfile.getParent() == null) {
-            System.out.printf("Skipping file %s%n", destfile);
+
+        if (!classFileName.endsWith(".class") || classFileName.equals("java/lang/Object.class")) {
+          File destfile = new File(classFileName);
+          // Copy non-.class files (and Object.class) unchanged (JDK 8).
+          // For JDK 9+ we do not copy as these items will be loaded from the original module file.
+          if (destfile.getParent() == null || BcelUtil.javaVersion > 8) {
+            if (verbose) System.out.printf("Skipping file %s%n", classFileName);
             continue;
           }
-          File dir = new File(dfile, destfile.getParent());
-          dir.mkdirs();
-          File destpath = new File(dir, destfile.getName());
+          File destParent = new File(dest_dir, destfile.getParent());
+          destParent.mkdirs();
+          File destpath = new File(destParent, destfile.getName());
           if (verbose) System.out.println("Copying Object or non-classfile: " + destpath);
-          copyStreams(jfile.getInputStream(entry), new FileOutputStream(destpath));
+          try (InputStream in = class_stream_map.get(classFileName)) {
+            Files.copy(in, destpath.toPath());
+          }
           continue;
         }
 
         // Get the binary for this class
-        InputStream is = jfile.getInputStream(entry);
+        InputStream is = class_stream_map.get(classFileName);
         JavaClass jc;
         try {
-          ClassParser parser = new ClassParser(is, entryName);
+          ClassParser parser = new ClassParser(is, classFileName);
           jc = parser.parse();
-        } catch (Exception e) {
-          throw new Error("Failed to parse entry " + entry, e);
+        } catch (Throwable e) {
+          throw new Error("Failed to parse classfile " + classFileName, e);
         }
-        classmap.put(jc.getClassName(), jc);
-      }
-
-      if (false) {
-        processClassFile(classmap, dfile, "sun.rmi.registry.RegistryImpl_Skel");
-        System.exit(0);
-      }
 
-      // Process each file read.
-      for (String classname : classmap.keySet()) {
-        JavaClass jc = classmap.get(classname);
+        // Instrument the class file.
         try {
-          processClassFile(classmap, dfile, classname);
+          instrumentClassFile(jc, dest_dir, classFileName, class_stream_map.size());
         } catch (Throwable e) {
-          throw new Error("Couldn't instrument " + classname, e);
+          throw new Error("Couldn't instrument " + classFileName, e);
         }
       }
+    } catch (Exception e) {
+      throw new Error(e);
+    }
+  }
 
-      // Create the DcompMarker class (used to identify instrumented calls)
-      ClassGen dcomp_marker =
+  /**
+   * Add abstract interface classes needed by DynComp runtime.
```

Comment:
Can you clarify this comment?  What are they added to?  Should the method be named "create" rather than "add"?

Reply:
changed comment to:
* Create abstract interface classes needed by DynComp runtime and write to destDir.